### PR TITLE
MBS-9904: Convert inline list of users from TT to React

### PR DESCRIPTION
**Pull request by TheFaR8**
https://tickets.metabrainz.org/browse/MBS-9904

Only a part of them (not fully converted yet)

### CODE REVIEW

File: root/user/privileged.tt [hunk 1]
```diff
@@ -1,15 +1,4 @@
...
-    [%- FOR editor=editors -%]
-      [%- link_entity(editor) -%]
-      [%- IF loop.count < editors.size -%], [%- END -%]
-    [%- END -%]
-  [% ELSE %]
-    [% l('No users found') %]
-  [% END %]
-</p>
-[% END %]
+[%- React.embed(c, 'components/UserInlineList', {editors => editors}) -%]
```
Comment (yvanzo) [inline root/user/privileged.tt]: The path must be relative from `root/`, see point 2. ii. at [Porting TT to React](https://github.com/metabrainz/musicbrainz-server/blob/master/HACKING.md#porting-tt-to-react), that is `user/components/UserInlineList` here.

File: root/user/components/UserInlineList.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+import * as React from 'react';
+
+import {l} from '../../static/scripts/common/i18n';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import commaOnlyList from '../../static/scripts/common/i18n/commaOnlyList';
+
+type Props = {|
+  +editors: EditorT,
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: `editors` is an array which is accessed through the `map` function on line 24.

Should be typed as:
```javascript
  +editors: $ReadOnlyArray<EditorT>,
```

File: root/user/components/UserInlineList.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+|};
+
+const UserInlineList = ({editors}: Props) => {
+  <p>
+    {editors.size ? (
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: `size` doesn’t exist for array, you mean `length`

File: root/user/components/UserInlineList.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      {commaOnList(
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: Typo, should be `commaOnlyList`

File: root/user/components/UserInlineList.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        editors.map((editor) => (
+          <EntityLink entity={editor} />
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: Should use `EditorLink` instead, with `editor` property

File: root/user/components/UserInlineList.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+import {l} from '../../static/scripts/common/i18n';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import commaOnlyList from '../../static/scripts/common/i18n/commaOnlyList';
+
+type Props = {|
+  +editors: EditorT,
+|};
+
+const UserInlineList = ({editors}: Props) => {
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: Opening bracket is not appropriate here as it should be used to enclose JavaScript expression whereas `<p>` is a JSX element. You can replace it with opening parenthesis if you like. (And update closing bracket on line 32 accordingly.)

File: root/user/components/UserInlineList.js (hunk 4 above)
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: Opening bracket is not appropriate here either, as you appropriately opened a bracket for JavaScript expressions as children on line 22 already. Just remove it (from line 23). (And remove closing bracket on line 28 accordingly.)

File: root/user/components/UserInlineList.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  <p>
+    {editors.size ? (
+      {commaOnList(
+        editors.map((editor) => (
+          <EntityLink entity={editor} />
+        )),
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: There is an extra closing parenthesis, as the following line should be the second parameter to call the `map` function. Just move the extra parenthesis to line 28.

File: root/user/components/UserInlineList.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
```
Comment (yvanzo) [inline root/user/components/UserInlineList.js]: You might want to add your own copyright notice here, or replace the current one, or not change anything. Note that this notice will be public and indexed by search engines.

File: root/user/components/UserInlineList.js (hunk 6 above)
Comment (TheFaR8) [reply root/user/components/UserInlineList.js]: Change `{` with `(` , right?

File: root/user/components/UserInlineList.js (hunk 7 above)
Comment (yvanzo) [reply root/user/components/UserInlineList.js]: Sorry, I was wrong here, it was correct. The first `)` closed the mapped anonymous function while the second `)` closes the `map` function.

**Comment by TheFaR8** [comment]
Thank you too for the suggested revisions :)

**Pull request merged**